sdks/java/io/influxdb/src/main/java/org/apache/beam/sdk/io/influxdb/InfluxDbIO.java

```diff
@@ -0,0 +1,662 @@
+import java.util.List;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.OkHttpClient;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDB.ConsistencyLevel;
+import org.influxdb.InfluxDBException;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import org.influxdb.dto.QueryResult.Result;
+import org.influxdb.dto.QueryResult.Series;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * IO to read and write from InfluxDB.
+ *
+ * <h3>Reading from InfluxDB</h3>
+ *
+ * <p>InfluxDB return a bounded collection of String as {@code PCollection<String>}. The String
+ * follows the line protocol
+ * (https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_tutorial/). To Configure
+ * the InfluxDB source, you whave to provide the connection URL, the credentials to connect to
+ * InfluxDB and the Database name
+ *
+ * <pre>{@code
+ *  pipeline.apply(
+ *    InfluxDB.read("https://influxdb", "userName", "password", "database")
+ *      .withQuery("select * from metric");
+ *      //Reads data based on the query from the InfluxDB
+ * }
+ *
+ * <p> The source also accepts optional configuration: {@code withRetentionPolicy()}  an</p>
+ *
+ *
+ * <h3>Writing to InfluxDB</h3>
+ *
+ * <p>InfluxDB sink supports writing data (as line protocol)  to InfluxDB
+ * To configure a InfluxDB sink, you must specify a URL {@code InfluxDBURL}, {@code userName}, {@code password}, {@code database}
+ * <pre>{@code
+ * pipeleine
+ *    .apply(...)
+ *    .appply(InfluxDB.write(https://influxdb", "userName", "password", "database")
+ *
+ * }
+ * </pre>
+ * </pre>
+ */
+@Experimental(Kind.SOURCE_SINK)
+public class InfluxDbIO {
+  private static final Logger LOG = LoggerFactory.getLogger(InfluxDbIO.class);
+
+  private static final String defaultRetentionPolicy = "autogen";
+
+  /** Disallow construction of utility class. */
+  private InfluxDbIO() {}
+
+  public static Write write(String influxDbUrl, String username, String password, String database) {
```

Comment:
Please, keep `write()` method without arguments for consistency with other IOs and add `withDataSourceConfiguration(DataSourceConfiguration conf)`, which will be required to configure data source, and `withDatabase(String)` to set database name (see `JdbcIO.write()` as an example).
So, for the user it would be something like:
```
InfluxDbIO.write().withDataSourceConfiguration(DataSourceConfiguration.create(...)).withDatabase("...");
```